src/picongpu/include/fields/laserProfiles/laserGaussianBeam.hpp

```diff
@@ -98,6 +124,12 @@ namespace picongpu
             // the radius of curvature of the beam's  wavefronts
             const float_X R_y = -FOCUS_POS * ( float_X(1.0) + ( y_R / FOCUS_POS )*( y_R / FOCUS_POS ) );
 
+            // initialize temporary variables
+            uint32_t m = 0;
+            float_X etrans = float_X(0.0);
+            float_X etrans_norm = float_X(0.0);
+            PMACC_CASSERT_MSG(MODENUMBER_must_be_smaller_than_number_of_entries_in_LAGUERREMODES_vector, MODENUMBER<LAGUERREMODES_t::dim);
+            for ( m = 0 ; m < MODENUMBER ; m++ ) etrans_norm += LAGUERREMODES[m];
```

Comment:
On the NaN problem: Since twelve modes have repeatedly worked before without generating NaNs, I suspect something is fishy with the vector `LAGUERREMODES` you used in your tests. If the sum of all coefficients is zero, the physical field is zero in the center focus at `FOCUS_POS`, so that our standard normalization of the center field to unity becomes ill-defined and produces NaNs later.